data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -307,6 +307,13 @@ private Map<String, String> generateIngestAndCommandHeaders(ClientRequestPropert
                 activityId, activityId, activityId, clientRequestId);
         headers.put("x-ms-activitycontext", activityContext);
 
+        // replace non-ascii characters in header values with '?'
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
+            String value = entry.getValue();
+            if (value != null) {
+                entry.setValue(value.replaceAll("[^\\x00-\\x7F]", "?"));
```

Comment:
Suggest using `[^\\x20-\\x7E]`, per https://www.asciitable.com/

Reply:
The point was to align this to the behavior of the c# SDK.
They don't touch other character, so we won't either.